File: src/main/java/TestGenerator.java
import edu.kruchon.natural.lang.parser.codegeneration.KotlinSource;
import edu.kruchon.natural.lang.parser.codegeneration.KotlinSourceGenerator;
import edu.kruchon.natural.lang.parser.syntax.TestParagraphSyntaxTreeParser;
import edu.kruchon.natural.lang.parser.syntax.Triplet;

import java.util.ArrayList;
import java.util.List;

public class TestGenerator {

    private final TestParagraphSyntaxTreeParser testParagraphSyntaxTreeParser = TestParagraphSyntaxTreeParser.INSTANCE;
    private final KotlinSourceGenerator kotlinSourceGenerator = KotlinSourceGenerator.INSTANCE;

    public List<KotlinSource> generate(String testScenario) {
        String[] paragraphs = testScenario.split("\\.");
        List<Triplet> triplets = new ArrayList<>(paragraphs.length);
        for (String paragraph : paragraphs) {
            triplets.add(testParagraphSyntaxTreeParser.parse(paragraph));
        }
        return kotlinSourceGenerator.generate(triplets);
    }
}

File: src/main/kotlin/TestGenerator.kt
import io.github.test.scenario.parser.generation.KotlinSource
import io.github.test.scenario.parser.generation.KotlinSourceGenerator
import io.github.test.scenario.parser.syntax.TestParagraphSyntaxTreeParser
import io.github.test.scenario.parser.syntax.Triplet

object TestGenerator {
    fun generate(testScenario: String): List<KotlinSource> {
        val paragraphs = testScenario.split(".")
        val triplets = mutableListOf<Triplet>()
        for (paragraph in paragraphs) {
            triplets.add(TestParagraphSyntaxTreeParser.parse(paragraph))
        }
        return KotlinSourceGenerator.generate(triplets)
    }
}